pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/defaultvalues/ATDefaultValuesTest.java

```diff
@@ -81,4 +105,96 @@ public void testSetDefaultValuesExceptionMessageWhenParticipantIdNotFound() {
                     e.getMessage().toUpperCase().contains("PARTICIPANT " + nonexistentParticipantId));
         }
     }
+
+    @Test
+    public void testSetDefaultValues() {
+        ParticipantDto participant = createParticipant();
+        String ddpParticipantId = participant.getDdpParticipantIdOrThrow();
+        ATDefaultValues atDefaultValues = new ATDefaultValues();
+        try {
+            atDefaultValues.generateDefaults(instanceName, ddpParticipantId);
+            Assert.fail("Should throw an exception when no ptp activities in ES");
+        } catch (Exception e) {
+            Assert.assertTrue(e instanceof ESMissingParticipantDataException);
+        }
+
+        List<Activities> activities = ReferralSourceServiceTest.getActivities();
+        ElasticTestUtil.addParticipantActivities(esIndex, activities, ddpParticipantId);
+
+        int fieldSettingsId = FieldSettingsTestUtil.createRadioFieldSettings(AT_PARTICIPANT_EXIT,
+                EXIT_STATUS,
+                "[{\"value\":\"0\",\"name\":\"Not Exited\",\"default\":true},"
+                        + "{\"value\":\"1\",\"name\":\"Exited\"}]",
+                ddpInstanceDto.getDdpInstanceId());
+        fieldSettingsIds.add(fieldSettingsId);
+
+        try {
+            boolean updated = atDefaultValues.generateDefaults(instanceName, ddpParticipantId);
+            Assert.assertTrue(updated);
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.fail("Exception from generateDefaults: " + e.getMessage());
+        }
+
+        ParticipantDataDao dataDao = new ParticipantDataDao();
+        List<ParticipantData> ptpDataList = dataDao.getParticipantDataByParticipantId(ddpParticipantId);
+        Assert.assertEquals(2, ptpDataList.size()); // 1 for exit status, 1 for genomic id
+        ptpDataList.forEach(ptpData -> {
+            String fieldType = ptpData.getRequiredFieldTypeId();
+            Map<String, String> dataMap = ptpData.getDataMap();
+            if (fieldType.equals(AT_PARTICIPANT_EXIT)) {
+                Assert.assertEquals("0", dataMap.get(EXIT_STATUS));
+            } else if (fieldType.equals(GENOME_STUDY_FIELD_TYPE)) {
+                Assert.assertTrue(dataMap.get(GENOME_STUDY_CPT_ID).startsWith(GENOMIC_ID_PREFIX));
+            } else {
+                Assert.fail("Unexpected field type: " + fieldType);
+            }
+        });
+
+        verifyElasticData(ddpParticipantId);
+
+        try {
+            // should not create additional genomic ids or exit statues
+            boolean updated = atDefaultValues.generateDefaults(instanceName, ddpParticipantId);
+            Assert.assertFalse(updated);
```

Comment:
Should we do a check into the underlying data directly to make sure we don't see multiple exit statuses or genomic ids?